oauth2-authorization-server/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/authorization/OAuth2AuthorizationServerConfigurer.java

```diff
@@ -144,6 +152,18 @@
 		return this;
 	}
 
+	/**
+	 * Sets the user info claims mapper.
+	 *
+	 * @param userInfoClaimsMapper the user info claims mapper
+	 * @return the {@link OAuth2AuthorizationServerConfigurer} for further configuration
+	 */
+	public OAuth2AuthorizationServerConfigurer<B> userInfoClaimsMapper(UserInfoClaimsMapper userInfoClaimsMapper) {
+		Assert.notNull(userInfoClaimsMapper, "userInfoClaimsMapper cannot be null");
+		this.getBuilder().setSharedObject(UserInfoClaimsMapper.class, userInfoClaimsMapper);
```

Comment:
With reduced visibility of the mapper class and moving it to this package, this would now wrap the given `Function<Object, OidcUserInfo>`, e.g.

```java
this.getBuilder().setSharedObject(UserInfoClaimsMapper.class, new OidcUserInfoClaimsMapper(userInfoClaimsMapper));
```

This is so the shared object mechanism works internally, but the class is not exposed in the public API.